generator/operation.go

```diff
@@ -739,8 +743,12 @@ func (b *codeGenOpBuilder) MakeHeaderItem(receiver, paramName, indexVar, path, v
 	res.Name = paramName
 	res.Path = path
 	res.Location = "header"
-	res.ValueExpression = valueExpression
+	res.ValueExpression = golang.MangleVarName(swag.ToGoName(valueExpression))
```

Comment:
why does this need to change, it was working afaik

Reply:
got naming problems with headers when starting generating collections. I try to gather evidence for the change (please wait)

Reply:
good catch: I left an intermediate experiment here. This is actually equivalent to the simple: swag.ToVarName().
Why this transform? Actually the idea came from issue #863 
Example for header: x-isAnOptionalHeader
With nested arrays constructs, I have an expressionValue at some point like 'x-IsAnOptionalHeader3I'
VarNaming it gives the more correct xIsAnOptionalHeader3I.
What is most important is that names are consistent with childs.